### PR TITLE
core/scopymainwindow_api: Implemented new API's

### PR DESCRIPTION
**Pull request by liealex**

### CODE REVIEW

**Review by IonutMuthi — CHANGES_REQUESTED** [review]

File: core/src/devicemanager.cpp [hunk 1]
```diff
@@ -40,16 +40,25 @@ void DeviceManager::addDevice(Device *d)
...
 	map[id] = d;
 	di->setParent(this);
+	QList<Plugin *> availablePlugins = di->plugins();
+	if(!m_plugins.isEmpty()) {
+		for(Plugin *p : qAsConst(availablePlugins)) {
+			if(!m_plugins.contains(p->name())) {
+				p->setEnabled(false);
+			}
+		}
+	}
```
Comment (andreidanila1) [inline core/src/devicemanager.cpp:50]: You can do this steps inside the api without changing the addDevice method.

File: core/include/core/scopymainwindow_api.h [hunk 2]
```diff
@@ -23,6 +28,19 @@ class SCOPY_CORE_EXPORT ScopyMainWindow_API : public ApiObject
 	Q_INVOKABLE void switchTool(QString toolName);
 	Q_INVOKABLE void runScript(QString scriptPath, bool exitApp = true);
 	Q_INVOKABLE void runScriptList(QStringList scriptPathList, bool exitApp = true);
+	Q_INVOKABLE void exit();
+	Q_INVOKABLE QStringList getTools();
+	Q_INVOKABLE QStringList getToolsPlugin(QString plugin);
```
[the remark this answers is not in the document]
Comment (liealex) [reply core/include/core/scopymainwindow_api.h]: solved

File: core/src/scopymainwindow_api.cpp [hunk 3]
```diff
@@ -196,4 +228,184 @@ bool ScopyMainWindow_API::sortByUUID(const QString &k1, const QString &k2)
...
+				toolList[i]->setRunning(flag);
+				toolList[i]->setRunEnabled(flag);
+			}
+		}
+	} else {
+		qWarning(CAT_SCOPY_API) << "Device with id " << devID << " is not available!";
+	}
+}
+
+void ScopyMainWindow_API::loadSetup(QString filename, QString path) { m_w->load(path + "/" + filename); }
```
[the remark this answers is not in the document]
Comment (liealex) [reply core/src/scopymainwindow_api.cpp]: solved

File: core/src/scopymainwindow_api.cpp [hunk 4]
```diff
@@ -196,4 +228,184 @@ bool ScopyMainWindow_API::sortByUUID(const QString &k1, const QString &k2)
...
+	return pluginList;
+}
+
+QStringList ScopyMainWindow_API::getPlugins(QString param, QString cat)
+{
+	QStringList pluginList = availablePlugins(param, cat, -1, nullptr);
+	return pluginList;
+}
+
+QStringList ScopyMainWindow_API::availablePlugins(QString param, QString cat, int idx, Device *dev)
```
[the remark this answers is not in the document]
Comment (liealex) [reply core/src/scopymainwindow_api.cpp]: idx removed

File: core/include/core/devicemanager.h [hunk 5]
```diff
@@ -62,6 +63,7 @@ private Q_SLOTS:
 	bool exclusive = false;
 	QStringList scannedDev;
 	QStringList connectedDev;
+	QList<QString> m_plugins;
```
[the remark this answers is not in the document]
Comment (liealex) [reply core/include/core/devicemanager.h]: removed

File: core/src/devicemanager.cpp (hunk 1 above)
Comment (liealex) [reply core/src/devicemanager.cpp]: steps removed from addDevice

File: core/include/core/devicemanager.h [hunk 6]
```diff
@@ -26,7 +26,8 @@ class SCOPY_CORE_EXPORT DeviceManager : public QObject
 public Q_SLOTS:
 
 	void addDevice(Device *d);
-	QString createDevice(QString category, QString param, bool async = true);
+	QString createDevice(QString category, QString param, bool async = true,
```
[the remark this answers is not in the document]
Comment (liealex) [reply core/include/core/devicemanager.h:29]: Yes, if you don't provide a plugin list then the device will be created with all its plugins.

[the remark this answers is not in the document]
Comment (liealex) [reply core/include/core/devicemanager.h:29]: Changed optional parameters position

**Review by IonutMuthi — APPROVED** [review]

**Pull request merged**